Add additional ProdCon updaters pointed at recent 2.1.1 build

dagood opened the pull request:
Upgrades buildtools to get new updaters implemented in https://github.com/dotnet/buildtools/pull/2047 and configures them, pointing at the recent 2.1.1 build.

I'm planning on porting the buildtools change from master to 2.1 once this gets merged, then porting this PR to source-build 2.1.

Ideally I'd use a master manifest rather than 2.1.1, but we don't have one. The source-build `master` appears to still be on 2.1.0 for now so this is still an upgrade.

#534

eerhardt commented on repos/dir.targets:
```diff
@@ -371,6 +371,7 @@
 
   <Target Name="GetProjectDirectory" Outputs="$(ProjectDirectory)" />
   <Target Name="GetOrchestratedManifestBuildName" Outputs="$(OrchestratedManifestBuildName)" />
+  <Target Name="GetOfficialBuildId" Outputs="OfficialBuildId" />
```
Does this need to be `$(OfficialBuildId)` ?

dagood commented:
In this PR I updated buildtools to latest `master` to get the new updaters, but this is incompatible with the 2.1.1 CoreFX commit. I reverted the BuildTools update to see if the submodule updates go through CI other than that.

Once https://github.com/dotnet/buildtools/pull/2063 is through, we'll have a 2.1 BuildTools that includes the updaters.

We're also at an awkward place where https://github.com/dotnet/source-build/pull/589 will update the Core repos to `master` soon. There aren't any ProdCon `master` builds to point at, so it will become impossible to meaningfully test auto-update config changes in source-build `master`.

dagood commented:
@crummel looks like I ran into why we need net46 builds? (https://github.com/dotnet/source-build/issues/592)

```
/mnt/j/workspace/dotnet_source-build/master/Ubuntu16.04_Release_prtest/src/sdk/src/Tasks/Microsoft.NET.Build.Extensions.Tasks/Microsoft.NET.Build.Extensions.Tasks.csproj :
error NU1202: Package Microsoft.Build.Framework 15.7.179 is not compatible with net46 (.NETFramework,Version=v4.6).
Package Microsoft.Build.Framework 15.7.179 supports: netstandard2.0 (.NETStandard,Version=v2.0) 
[/mnt/j/workspace/dotnet_source-build/master/Ubuntu16.04_Release_prtest/src/sdk/sdk.sln] [/mnt/j/workspace/dotnet_source-build/master/Ubuntu16.04_Release_prtest/repos/sdk.proj]
```

I'll revert the upgrade entirely for now, and just get the configuration in.
